src/Tools/dotnet-monitor/ConfigurationJsonWriter.cs

```diff
@@ -202,28 +202,72 @@ private void ProcessSection(IConfigurationSection section, bool includeChildSect
             //If we do not traverse the child sections, the caller is responsible for creating the value
             if (includeChildSections && children.Any())
             {
-                using (new JsonObjectContext(_writer))
+                bool isSequentialIndices = CheckForSequentialIndices(children);
+
+                bool parentIsCR = section.Key.Equals(ConfigurationKeys.CollectionRules);
+
+                if (isSequentialIndices && !parentIsCR)
                 {
-                    foreach (IConfigurationSection child in children)
+                    if (children.First().GetChildren().Any())
```

Comment:
I don't understand this if check? isn't children.First the first element of the array?

Reply:
Yep; this check is actually to see if there are "grandchildren". To get rid of the array indices, we jump over them to their children and use those as new keys. However, if the children are values (leaf nodes), then the `else` condition has us write out the value. It's the same fundamental idea as the check on Line 203 to see if there are any children, except we do this specifically for array indices.

Reply:
In terms of only checking the first child, given that all elements in the array should be the same type if one has children they should all in theory have children...I'm trying to think if there's an edge case where this would be a problem